spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -157,11 +161,20 @@ public int getWriteBufferSize() {
 	 */
 	@Override
 	public <T> Flux<T> saveAll(Publisher<T> instances) {
-		Flux<List<T>> inputs = Flux.from(instances).bufferTimeout(
-				this.writeBufferSize, this.writeBufferTimeout);
-
-		return ObservableReactiveUtil.streamingBidirectionalCall(
-				this::openWriteStream, inputs, this::buildWriteRequest);
+		return Mono.subscriberContext().flatMapMany(ctx -> {
```

Comment:
It makes sense to externalize this code into a connection utility that checks `TransactionSynchronizationManager.forCurrentTransaction()` instead of using `TransactionContext` directly.